internal/elasticsearch/security/role.go

```diff
@@ -258,6 +263,13 @@ func resourceSecurityRolePut(ctx context.Context, d *schema.ResourceData, meta i
 	}
 	var role models.Role
 	role.Name = roleId
+
+	// Add description to the role
+	if v, ok := d.GetOk("description"); ok {
```

Comment:
Can we wrap this in a version check similarly to https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/security/role.go#L364-L366